Pioneer Build Error

Strangely getting a build error locally, whereas github workflow and netlify checks are passing.

`yarn workspace pioneer build`

on development branch

```
Version 3.7.2
joy-proposals/src/Proposal/ChooseProposalType.tsx:42:55 - error TS2339: Property 'category' does not exist on type 'never'.

42             .filter(typeInfo => !category || typeInfo.category === category)
                                                         ~~~~~~~~

joy-utils/src/react/context/transport.tsx:5:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

5 import Transport from "../../transport";
         ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:34:13 - error TS7006: Parameter 'unsubscribe' implicitly has an 'any' type.

34       .then(unsubscribe => {
               ~~~~~~~~~~~

joy-utils/src/react/hooks/useTransport.tsx:2:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

2 import Transport from "../../transport";
         ~~~~~~~~~


Found 4 errors.
```


```sh
MokoBook:joystream mokhtar$ yarn --version
1.16.0
MokoBook:joystream mokhtar$ node --version
v12.17.0
```

Attempting to fix in #514 Brings up a few other errors..

```
joy-proposals/src/forms/EvictStorageProviderForm.tsx:42:93 - error TS2339: Property 'storageProviders' does not exist on type 'Transport'.

42   const [storageProviders /* error */, , loading] = usePromise<AccountId[]>(() => transport.storageProviders.providers(), []);
                                                                                               ~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupLeadForm.tsx:81:21 - error TS2339: Property 'council' does not exist on type 'Transport'.

81     () => transport.council.membersExceptCouncil(),
                       ~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupLeadForm.tsx:85:21 - error TS2339: Property 'contentWorkingGroup' does not exist on type 'Transport'.

85     () => transport.contentWorkingGroup.currentLead(),
                       ~~~~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetContentWorkingGroupMintCapForm.tsx:9:79 - error TS2339: Property 'contentWorkingGroup' does not exist on type 'Transport'.

9   const [ mintCapacity, error, loading ] = usePromise<number>(() => transport.contentWorkingGroup.currentMintCap(), 0);
                                                                                ~~~~~~~~~~~~~~~~~~~

joy-proposals/src/forms/SetCouncilParamsForm.tsx:72:99 - error TS2339: Property 'council' does not exist on type 'Transport'.

72   const [ councilParams, error, loading ] = usePromise<ElectionParameters | null>(() => transport.council.electionParameters(), null);
                                                                                                     ~~~~~~~

joy-proposals/src/forms/SetMaxValidatorCountForm.tsx:36:63 - error TS2339: Property 'validators' does not exist on type 'Transport'.

36   const [validatorCount] = usePromise<number>(() => transport.validators.maxCount(), NaN);
                                                                 ~~~~~~~~~~

joy-proposals/src/forms/SetStorageRoleParamsForm.tsx:82:78 - error TS2339: Property 'storageProviders' does not exist on type 'Transport'.

82   const [params] = usePromise<IStorageRoleParameters | null>(() => transport.storageProviders.roleParameters(), null);
                                                                                ~~~~~~~~~~~~~~~~

joy-proposals/src/Proposal/Body.tsx:47:21 - error TS2339: Property 'members' does not exist on type 'Transport'.

47     () => transport.members.memberProfile(memberId),
                       ~~~~~~~

joy-proposals/src/Proposal/ChooseProposalType.tsx:23:70 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

23   const [proposalTypes, error, loading] = usePromise(() => transport.proposals.proposalsTypesParameters(), []);
                                                                        ~~~~~~~~~

joy-proposals/src/Proposal/ChooseProposalType.tsx:42:55 - error TS2339: Property 'category' does not exist on type 'never'.

42             .filter(typeInfo => !category || typeInfo.category === category)
                                                         ~~~~~~~~

joy-proposals/src/Proposal/ProposalPreviewList.tsx:55:84 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

55   const [proposals, error, loading] = usePromise<ParsedProposal[]>(() => transport.proposals.proposals(), []);
                                                                                      ~~~~~~~~~

joy-proposals/src/Proposal/VotingSection.tsx:58:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

58     () => transport.proposals.voteByProposalAndMember(proposalId, memberId),
                       ~~~~~~~~~

joy-utils/src/react/context/transport.tsx:5:8 - error TS2613: Module '"/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' has no default export. Did you mean to use 'import { Transport } from "/Users/mokhtar/joystream/joystream/pioneer/packages/joy-utils/src/Transport"' instead?

5 import Transport from "../../transport";
         ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:14:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

14     () => transport.proposals.proposalById(id),
                       ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:19:21 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

19     () => transport.proposals.votes(id),
                       ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:33:15 - error TS2339: Property 'proposals' does not exist on type 'Transport'.

33     transport.proposals.subscribeProposal(id, refreshProposalData)
                 ~~~~~~~~~

joy-utils/src/react/hooks/proposals/useProposalSubscription.tsx:34:13 - error TS7006: Parameter 'unsubscribe' implicitly has an 'any' type.

34       .then(unsubscribe => {
               ~~~~~~~~~~~


Found 17 errors.
```

--- a/.github/workflows/pioneer-pr.yml
+++ b/.github/workflows/pioneer-pr.yml
@@ -19,7 +19,6 @@ jobs:
         yarn install --frozen-lockfile
         yarn workspace pioneer run build:code
 
-jobs:
   pioneer_build_code_osx:
     name: (MacOS) Build Code
     runs-on: macos-latest

--- a/.github/workflows/pioneer-pr.yml
+++ b/.github/workflows/pioneer-pr.yml
@@ -1,10 +1,28 @@
-name: Pioneer-PR
+name: Pioneer
 on: [pull_request, push]
 
 jobs:
-  pioneer_build_code:
-    name: Pioneer Build Code
-    runs-on: [ubuntu-latest, macos-latest]
+  pioneer_build_code_ubuntu:
+    name: (Ubuntu) Build Code
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [12.x]
+    steps:
+    - uses: actions/checkout@v1
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: build
+      run: |
+        yarn install --frozen-lockfile
+        yarn workspace pioneer run build:code
+
+jobs:
+  pioneer_build_code_osx:
+    name: (MacOS) Build Code
+    runs-on: macos-latest
     strategy:
       matrix:
         node-version: [12.x]

--- a/.github/workflows/pioneer-pr.yml
+++ b/.github/workflows/pioneer-pr.yml
@@ -2,8 +2,8 @@ name: Pioneer
 on: [pull_request, push]
 
 jobs:
-  pioneer_build_code_ubuntu:
-    name: (Ubuntu) Build Code
+  pioneer_build_ubuntu:
+    name: Ubuntu Build
     runs-on: ubuntu-latest
     strategy:
       matrix:
@@ -17,10 +17,10 @@ jobs:
     - name: build
       run: |
         yarn install --frozen-lockfile
-        yarn workspace pioneer run build:code
+        yarn workspace pioneer run build
 
-  pioneer_build_code_osx:
-    name: (MacOS) Build Code
+  pioneer_build_osx:
+    name: MacOS Build
     runs-on: macos-latest
     strategy:
       matrix:
@@ -34,21 +34,4 @@ jobs:
     - name: build
       run: |
         yarn install --frozen-lockfile
-        yarn workspace pioneer run build:code
-
-  pioneer_build_i18n:
-    name: Pioneer Build i18n
-    runs-on: [ubuntu-latest, macos-latest]
-    strategy:
-      matrix:
-        node-version: [12.x]
-    steps:
-    - uses: actions/checkout@v1
-    - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v1
-      with:
-        node-version: ${{ matrix.node-version }}
-    - name: build
-      run: |
-        yarn install --frozen-lockfile
-        yarn workspace pioneer run build:i18n
+        yarn workspace pioneer run build

--- a/.github/workflows/pioneer-pr.yml
+++ b/.github/workflows/pioneer-pr.yml
@@ -4,7 +4,7 @@ on: [pull_request, push]
 jobs:
   pioneer_build_code:
     name: Pioneer Build Code
-    runs-on: ubuntu-latest
+    runs-on: [ubuntu-latest, macos-latest]
     strategy:
       matrix:
         node-version: [12.x]
@@ -21,7 +21,7 @@ jobs:
 
   pioneer_build_i18n:
     name: Pioneer Build i18n
-    runs-on: ubuntu-latest
+    runs-on: [ubuntu-latest, macos-latest]
     strategy:
       matrix:
         node-version: [12.x]

--- a/pioneer/packages/joy-utils/src/react/context/transport.tsx
+++ b/pioneer/packages/joy-utils/src/react/context/transport.tsx
@@ -2,7 +2,7 @@ import React, { createContext, useContext } from "react";
 import { ApiContext } from "@polkadot/react-api";
 import { ApiProps } from "@polkadot/react-api/types";
 
-import { Transport } from "../../transport";
+import Transport from "../../transport/index";
 
 export const TransportContext = createContext<Transport>((null as unknown) as Transport);
 

--- a/pioneer/packages/joy-utils/src/react/context/transport.tsx
+++ b/pioneer/packages/joy-utils/src/react/context/transport.tsx
@@ -2,7 +2,7 @@ import React, { createContext, useContext } from "react";
 import { ApiContext } from "@polkadot/react-api";
 import { ApiProps } from "@polkadot/react-api/types";
 
-import Transport from "../../transport";
+import { Transport } from "../../transport";
 
 export const TransportContext = createContext<Transport>((null as unknown) as Transport);
 

--- a/pioneer/packages/joy-utils/src/react/hooks/useTransport.tsx
+++ b/pioneer/packages/joy-utils/src/react/hooks/useTransport.tsx
@@ -1,5 +1,5 @@
 import { useContext } from 'react';
-import Transport from "../../transport";
+import { Transport } from "../../transport";
 import { TransportContext } from "../context";
 
 export default function useTransport() {

--- a/pioneer/packages/joy-utils/src/react/hooks/useTransport.tsx
+++ b/pioneer/packages/joy-utils/src/react/hooks/useTransport.tsx
@@ -1,5 +1,5 @@
 import { useContext } from 'react';
-import { Transport } from "../../transport";
+import Transport from "../../transport/index";
 import { TransportContext } from "../context";
 
 export default function useTransport() {

